rust/parquet/src/arrow/converter.rs

```diff
@@ -128,7 +128,10 @@ pub struct Utf8ArrayConverter {}
 
 impl Converter<Vec<Option<ByteArray>>, StringArray> for Utf8ArrayConverter {
     fn convert(source: Vec<Option<ByteArray>>) -> Result<StringArray> {
-        let mut builder = StringBuilder::new(source.len());
+        let mut builder = StringBuilder::with_capacity(
+            source.len(),
+            source.len() * std::mem::size_of::<ByteArray>(),
```

Comment:
This might be a touch on the pessimistic side, but appears to be the same approximation made in `ComplexObjectArrayReader`. It would be nice to somehow feed the data from `ComplexObjectArrayReader` through to the converter, but this was a non-trivial change so I avoided it.

Reply:
Yeah this could be overly pessimistic. A more accurate way is to calculate the sum of all lens. It may be worth it.

Reply:
Having revisited this, not only is it pessimistic - it's plain wrong :). The ByteArray type here is just a wrapper pointer struct and its size in memory has no semantic value. This is fixed in the latest version which computes the correct array capacity.